python/mpt/src/mpt/trie_diff.cairo

```diff
@@ -1430,59 +1579,102 @@ func _compute_left_branch_on_right_extension_node{
     main_trie_end: AddressAccountDiffEntry*,
     storage_tries_end: StorageDiffEntry*,
 }(
-    left: BranchNode,
-    right: ExtensionNode,
+    subnodes: Subnodes,
```

Comment:
why changing the signature here?

Reply:
to fit the symmetrical impl. of the same logic

Reply:
i'd revert that change unless strictly necessary because it breaks the pattern of all helper functions being left node vs right node

Reply:
also is there a check_branch_node missing here?

Reply:
> i'd revert that change unless strictly necessary because it breaks the pattern of all helper functions being left node vs right node

it's the current signature of `_compute_left_extension_node_diff_on_right_branch_node`

Reply:
i'm simply making _compute_left_branch_node_diff_on_right_extension_node a symmetrical helper

Reply:
> also is there a check_branch_node missing here?

i dont think so, since `_compute_left_branch_node_diff_on_right_extension_node` is called in `_left_is_branch_node` which performs the check

Reply:
right 👍